runtime-modules/council/src/lib.rs

```diff
@@ -793,11 +805,13 @@ impl<T: Trait> EnsureChecks<T> {
         let candidate = Candidates::<T>::get(membership_id);
 
         // prevent user from releasing candidacy stake during election
-        if candidate.cycle_id == AnnouncementPeriodNr::get() {
-            match Stage::<T>::get().stage {
-                CouncilStage::Idle => (),
-                _ => return Err(Error::StakeStillNeeded),
+        if candidate.cycle_id == AnnouncementPeriodNr::get()
```

Comment:
Please use `matches` macro, as prior review said, this is needlessly verbose, macro gives pure variant condition test.

Reply:
Done; macro used. `if candidate.cycle_id == AnnouncementPeriodNr::get()
            && !matches!(Stage::<T>::get().stage, CouncilStage::Idle)
        {`